modules/library/render/src/main/java/org/geotools/renderer/lite/StreamingRenderer.java

```diff
@@ -2326,18 +2325,49 @@ protected GridCoverage2D readCoverage(
                                 GridGeometry2D readGG)
                                 throws IOException {
                             Interpolation interpolation = getRenderingInterpolation(layer);
-                            GridCoverageReaderHelper helper;
+                            RenderingHints interpolationHints =
+                                    new RenderingHints(JAI.KEY_INTERPOLATION, interpolation);
+                            final GridCoverageRenderer gcr;
                             try {
-                                helper =
-                                        new GridCoverageReaderHelper(
+                                // Use the original screenSize
```

Comment:
Uh, this is going vertical (format wise)... could you use a "convert anonymous to inner" and move this block to its own inner class?